pandas/core/missing.py

```diff
@@ -621,3 +622,24 @@ def fill_zeros(result, x, y, name, fill):
             result = result.reshape(shape)
 
     return result
+
+
+def maybe_fill(obj, value, errors):
+    """
+    fillna error coercion routine, returns whether or not to continue.
+    """
+    from pandas import Series
+    if isinstance(obj, Series):
+        if errors is None or errors == 'coerce':
+            return True
```

Comment:
Previously, I provided `coerce` as the default kwarg and passed that on through the `fillna` method stack. However, whilst Sparse and Categorical series accept an `errors` param, but raise immediately if it is not `None` (because it's not implemented/doesn't make sense for them). So the default (`coerce`) was causing those routines to raise&mdash;and a *lot* of test failures.

I remade the default param to `None` and introduced this modified check to address this concern.

How should this be handled instead?